main.tf

```
@@ -432,6 +448,17 @@ resource "aws_lambda_function_url" "this" {
   }
 }
 
+resource "aws_lambda_function_recursion_config" "this" {
+  count = local.create && var.recursive_loop == "Allow" ? 1 : 0
+
+  function_name  = var.function_name
```

```suggestion
  function_name  = aws_lambda_function.this[0].function_name
```

Then we don't need `depends_on`.